### PR TITLE
Openapi param serialization tutorial

### PR DESCRIPTION
**Pull request by charjr**

### CODE REVIEW

File: src/_guides/openapi/specification/v3.1/understanding-structure/parameter-serialization.md [hunk 1]
```diff
@@ -1,6 +1,261 @@
...
 title: Parameter Serialization
-authors: phil
-excerpt: 
+authors: John Charman
+excerpt: Follow through examples of how your parameters will be serialized.
 date: 2024-07-04
 ---
+
+[Parameters](parameters.md) not only define what inputs your API accepts, they also define the format your API expects to receive them in, i.e. how you would like it serialized.
+
```
Comment (philsturgeon) [inline src/_guides/openapi/specification/v3.1/understanding-structure/parameter-serialization.md:9]: Perhaps a quick explainer about why youd need this. "For example most parameters in query strings or headers are a string or integer, but did you know you can accept arrays, and even objects?" with examples showing whats up, before we delve into the keywords explode and styl.e

File: src/_guides/openapi/specification/v3.1/understanding-structure/parameter-serialization.md [hunk 2]
```diff
@@ -1,6 +1,261 @@
...
+`explode` defines whether parameters should be broken into logical components.
+
+It takes a boolean value:
+- If `true`; a parameter with multiple values will be serialized as if each of its values were separate parameters.  
+  - What separates each parameter is determined by the `style`.
+- If `false`; a parameter is a single parameter, regardless of how many values it has.
+
+In practice, this means only parameters of `type:array` or `type:object` are affected by `explode`.
+- For an array, each value becomes its own parameter.
+- For an object, each key-value pair is concatenated into into its own parameter as "key=value".
```
Comment (philsturgeon) [inline src/_guides/openapi/specification/v3.1/understanding-structure/parameter-serialization.md]:
```suggestion
- For an object, each key-value pair is concatenated into its own parameter as "key=value".
```

File: src/_guides/technical-writing/incorporating-api-documentation-guidelines-into-your-api-style-guide.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
+---
```
Comment (philsturgeon) [inline src/_guides/technical-writing/incorporating-api-documentation-guidelines-into-your-api-style-guide.md]: What is this new file, is this an accident?

**Review by philsturgeon — COMMENTED** [review]
This is all looking pretty good, very in depth, and you were definitely the right person to get to help! 

If you could try and pop a few "real world" examples in a little bit, that would really help. Even if they are a bit contrived, an example HTTP request and the OpenAPI source to match it should really help take this from "an easier version of the spec" to a tutorial somebody could use to make this complex topic simple using real world examples they can copy and paste to get started.

Max out at 2 more hours on this and see how far you get.

**Comment by philsturgeon** [comment]
Oh sorry about all the conflicts @charjr.

If you do a `git pull --rebase origin openapi-tutorials` and `git push -f <your repo> <branch>` it should calm down.

File: src/_guides/technical-writing/incorporating-api-documentation-guidelines-into-your-api-style-guide.md (hunk 3 above)
Comment (charjr) [reply src/_guides/technical-writing/incorporating-api-documentation-guidelines-into-your-api-style-guide.md]: I think I may have accidentally rebased off the wrong branch at first. So it added a file someone else may have been working on.

**Comment by charjr** [comment]
> Oh sorry about all the conflicts @charjr.
> 
> If you do a `git pull --rebase origin openapi-tutorials` and `git push -f <your repo> <branch>` it should calm down.

I've rebased the branch. Is there a time frame you want this complete by?

File: src/_guides/technical-writing/incorporating-api-documentation-guidelines-into-your-api-style-guide.md (hunk 3 above)
Comment (charjr) [reply src/_guides/technical-writing/incorporating-api-documentation-guidelines-into-your-api-style-guide.md]: Rebasing resolved this.

File: src/_guides/openapi/specification/v3.1/understanding-structure/parameter-serialization.md (hunk 2 above)
Comment (charjr) [reply src/_guides/openapi/specification/v3.1/understanding-structure/parameter-serialization.md]: Thank you, fixed.

**Comment by philsturgeon** [comment]
This is brilliant, thank you. I'll give it a little tweaking on my end as its too trivial to bother switching context for.

**Pull request merged**